cpp/src/arima/batched_arima.cu

```diff
@@ -125,13 +141,9 @@ void predict(cumlHandle& handle, const double* d_y, int batch_size, int n_obs,
                            d_y_fc[num_steps * bid + i];
                        }
                      });
-
-    allocator->deallocate(d_y_fc, num_steps * batch_size * sizeof(double),
-                          stream);
+    /// TODO: 2D copy kernel?
```

Comment:
We could avoid it by passing strides/offsets to the Kalman filter and the un-differencing kernel. This code will need some refactoring eventually.

Reply:
I think passing strides/offsets is a fine way of programming GPUs. Libraries like `cublas` allow it (eg: `lda` parameter to `gemm`) specifically to avoid having to create another copy of the inputs or outputs.

Reply:
Agreed